src/util/StringUtils.h

```diff
@@ -244,6 +260,35 @@ constexpr bool constantTimeEquals(std::string_view view1,
   return impl(toVolatile(view1), toVolatile(view2));
 }
 
+// _________________________________________________________________________
+template <typename Range, typename TranslationFunction>
+static std::string listToString(Range&& r,
+                                TranslationFunction translationFunction,
+                                std::string_view listItemSeparator) {
+  if (r.empty()) {
+    return "";
+  }
+
+  std::ostringstream stream;
+
+  /*
+  TODO<C++23>: This can be a combination of `std::views::transform` and
+  `std::views::join_with`. After that, we just have to insert all the elements
+  of the new view into the stream.
+  */
+  forEachExcludingTheLastOne(
+      AD_FWD(r),
+      [&stream, &translationFunction,
+       &listItemSeparator](const auto& listItem) {
+        stream << translationFunction(listItem) << listItemSeparator;
+      },
+      [&stream, &translationFunction](const auto& listItem) {
+        stream << translationFunction(listItem);
+      });
+
+  return stream.str();
+}
```

Comment:
And of course, write unit tests for this:)

Reply:
Done, see [b54efe5](https://github.com/ad-freiburg/qlever/pull/1010/commits/b54efe52ad05cba4d89a336d7576ad86f524847d).

Can you think of anymore tests for the function?